Sources/Purchasing/Purchases/TransactionPoster.swift

```diff
@@ -162,15 +231,42 @@ final class TransactionPoster: TransactionPosterType {
 
 }
 
-/// Async extension
+// MARK: - Extensions
+
+extension TransactionPosterResult {
+
+    typealias CustomerInfoFetcher = (@escaping @Sendable (Result<CustomerInfo, BackendError>) -> Void) -> Void
+
+    /// Converts the `TransactionPosterResult` into `Result<CustomerInfo, BackendError>`
+    /// by using `customerInfoFetcher` if the result is `.alreadyPosted`.
+    func toResult(
```

Comment:
The comment is clear but I'm concerned the name of the function doesn't indicate that this might initiate a fetch of the customer info... Maybe we could rename to `toResultOrFetchCustomerInfo`? Or something along those lines?